### DISCUSSION TITLE
Unsure about differences in preprocessing of volumes used for topup

### DISCUSSION THREAD

**Discussion by psadil** [PennLINC/qsiprep#401] [Q&A] [answered]
I'm working with a dataset that has dedicated fieldmaps collected with reversed phase encoding directions. It's my impression that the `topup` node, when in the pipeline, builds the input from combinations of the dedicated fieldmaps and b=0 volumes from the DWI (currently without considering the `--prefer-dedicated-fieldmaps` option, as per #321). However, when `QSIprep` is configured to cleanup the DWI scan through denoising, ungrining, and bias correction, it seems like that cleanup only happens for the volumes sourced from the DWI scan itself, and not the fieldmaps. So, the inputs to `topup` comprise both raw and preprocessed volumes. 

Would it be feasible to avoid this difference, so that all volumes input to `topup` receive the same treatment? For example, two ways to do this would be to restore the functionality of the `--prefer-dedicated-fieldmaps` option (so that all inputs to `topup` are raw), or to also preprocess the fieldmaps prior to `topup`. 

Thanks!

<details>
<summary>QSIprep run command</summary>

```
singularity run -e --nv docker://pennbbl/qsiprep:0.14.2 \
  "${BIDS}" . participant \
  --output-resolution 1.7 --bids-filter-file bids.json \
  --denoise-method patch2self --unringing-method mrdegibbs \
  --hmc_model eddy --eddy_config eddy_params.json \
  --fs-license-file license.txt --participant_label "${subject_id}" -vvv​
```

</details>

**Comment by mattcieslak** [comment, marked as answer]
Sorry for the super late reply - as of 0.16.1 the images used for topup are forced to have the same level of processing. This ended up being important for ABCD data

**Comment by psadil** [comment]
That's great! Thanks for the response (and patch). Just to be clear, does this mean that these updates will _not_ be available in the version that currently has a release candidate, [0.16.0RC3](https://github.com/PennLINC/qsiprep/releases/tag/0.16.0RC3), but that instead I should expect to see them in the subsequent version?
